src/OrchardCore.Modules/OrchardCore.Lists/Views/ListPartDetailAdmin.cshtml

```diff
@@ -15,8 +19,21 @@
     {
         <script asp-src="~/OrchardCore.Lists/Scripts/ci-list-ordering.min.js" debug-src="~/OrchardCore.Lists/Scripts/ci-list-ordering.js" asp-name="ci-list-ordering" at="Foot" depends-on="sortable, jQuery"></script>
     }
-}
 
+    var authorizedIndexes =
+        await Task.WhenAll(
+            Model.ContainedContentTypeDefinitions.
+                Select(async contentTypeDefinition =>
+                    await AuthorizationService.AuthorizeAsync(
+                        User,
+                        CommonPermissions.EditContent,
+                        ContentManager.NewAsync(contentTypeDefinition.Name))));
+
+    var authorizedContentTypeDefinitions =
+        Model.ContainedContentTypeDefinitions.
+            Where(contentTypeDefinition =>
+                authorizedIndexes[Model.ContainedContentTypeDefinitions.ToList().IndexOf(contentTypeDefinition)]);
+}
```

Comment:
Can this be done easier? I feel like I am overthinking this.

Reply:
See my comments.

Reply:
So after rethinking my life choices and my code too. And I think it would be best if I just use "foreach" here like in the AdminController.